src/shogun/base/SGObject.cpp

```diff
@@ -829,30 +851,13 @@ void CSGObject::subscribe_to_parameters(ParameterObserverInterface* obs)
 void CSGObject::observe_scalar(
     const int64_t step, const std::string& name, const Any& value)
 {
+	if (!m_enable_emit)
```

Comment:
rx will take care of this internally, no need to do ourselves:
https://github.com/Reactive-Extensions/RxCpp/blob/master/Rx/v2/src/rxcpp/subjects/rx-subject.hpp#L185